scripts/sql_snippets/create_new_tables.sql

```diff
@@ -22,24 +22,26 @@ CREATE TABLE scientificname (
     "taxonomyId" integer REFERENCES taxonomy(id), -- Can be null if no match
     "deprecatedTaxonId" integer REFERENCES taxon(id),
     "scientificName" character varying(255) NOT NULL, -- as appear in the old "taxon" table
-    "authorship" character varying(255), -- as appear in the old "taxon" table,
+    "authorship" character varying(255), -- as appear in the old "taxon" table or in annexes,
     -- !! the following field are attributes of the match process !!
     "lastMatched" timestamp with time zone, -- when was a GBIF match last attempted?
     "matchConfidence" smallint,
-    "matchType" gbifmatchtype
+    "matchType" gbifmatchtype,
+    CONSTRAINT scn_auth UNIQUE("scientificName", "authorship")
 );
+CREATE UNIQUE INDEX scn_auth_not_null ON scientificname("scientificName", ("authorship" IS NULL))
```

Comment:
This works*, but I am wondering if we cannot simplify it to:

```
CREATE UNIQUE INDEX scn_auth_not_null ON scientificname("scientificName")
WHERE "authorship" IS NULL;
```

*: works: tested with simple INSERTS (reject second insert if author is null or not). I think the ON CONFLIC-returns None issue that you mentioned as the main problem is a different one - looking into it!

Reply:
Thanks, I will test it. It makes sense.